src/Board.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Board.h"
+
+void Board::setBoard(const std::string& board) {
+	m_piece.resize(8); // resize the 2D vector
+	for (auto& row : m_piece) row.resize(8); // resize each row
+
+	// loop through the board string and create the corresponding piece objects
+	for (int i = 0; i < board.size(); ++i) {
+		int x = i % 8; // calculate x and y indices for the spot
+		int y = i / 8;
+		m_piece[y][x] = Factory<Piece>::create(board[i]);
+	}
+	setKingsPositions();
+}
+
+void Board::setKingsPositions() {
+	for (int i = 0; i < BoardSize; i++)
+		for (int j = 0; j < BoardSize; j++) 
+			if (m_piece[i][j] && typeid(*m_piece[i][j]) == typeid(King)) {
+				m_piece[i][j]->isWhite() ? setKingPos({ i + 1, j + 1 }, true) :
```

Comment:
You could make this line of code more readable, with something like this: 
```cpp
bool is_king_white = m_piece[i][j]->isWhite();
setKingPos({i+1,j+1}, is_king_white);
```
Reduces the length of the line and the duplication in it